airflow/utils/log/gcs_task_handler.py

```diff
@@ -129,7 +129,7 @@ def gcs_read(self, remote_log_location):
         :type remote_log_location: str (path)
         """
         bkt, blob = self.parse_gcs_url(remote_log_location)
-        return self.hook.download(bkt, blob).decode()
+        return self.hook.download(bkt, blob).decode('utf-8')
```

Comment:
The default encoding of `.decode()` is already "utf-8". Please refer to https://docs.python.org/3/library/stdtypes.html#bytes.decode

So `.decode('utf-8')` is no difference from `.decode()`.

Reply:
> So .decode('utf-8') is no difference from .decode().
I have a different opinion.  Airflow support Python 2.7, 3.6=> ([Source](https://github.com/apache/airflow/blob/master/setup.py#L424))

In documentation for Python 2.7, you can read a fragment:
> Python’s default encoding is the ‘ascii’ encoding.
Source: https://docs.python.org/2/howto/unicode.html

It is also worth quoting another fragment
> str.decode([encoding[, errors]])
> Decodes the string using the codec registered for encoding. encoding defaults to the default string encoding.

Taking into account the quotations above, the change proposed here changes the behavior of the program.

Reply:
Exactly, I get encoding error with python 2.7, forgot to mention that

Reply:
I see. Thanks @mik-laj and @RasPavel for the clarification